freight/hooks/github.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+
+__all__ = ['GitHubHooks']
+
+from flask import current_app, request, Response
+
+from .base import Hook
+from freight.testutils.client import AuthenticatedTestClient
+
+
+class GitHubHooks(Hook):
+    def ok(self):
+        return Response()
+
+    def deploy(self, app, env):
+        payload = request.get_json()
+        event = request.headers.get('X-GitHub-Event')
+
+        if event != 'push':
+            # Gracefully ignore everything except push events
+            return self.ok()
+
+        default_ref = app.get_default_ref(env)
+        ref = payload['ref']
+
+        if ref != 'refs/heads/{}'.format(default_ref):
+            return self.ok()
+
+        head_commit = payload['head_commit']
+        committer = head_commit['committer']
+
+        client = AuthenticatedTestClient(
```

Comment:
This whole way to trigger a deploy feels a little weird, but I understand that it's the only way.